### PR TITLE
Editor: Update pending toggle logic to respond to user action

### LINKED ISSUE #4285: Editor: disabling pending does not change toggle until manually click 'save'
#### Steps to reproduce
1. Starting at URL: https://wordpress.com/post (new post)
2. Add content
3. Click draft to show more options
4. Enable pending review - toggle changes immediately
5. Click save - status changes to pending
6. Disable pending review - when clicked toggle doesn't change

#### What I expected

Toggle should change when clicked

#### What happened instead

* Toggle registers the click, but only changes when clicking 'save' which is quite confusing
* Should the auto-save kick in when this toggle is changed either direction? It doesn't, user has to manually click save each time

#### Browser / OS version

Chrome OSX

#### Screenshot / Video

pending status

### PR DESCRIPTION
**Pull request by jeremeylduvall**
This is meant to fix #4285. Currently, we're using two checks to determine if the pending toggle should be active:

1. `this.props.savedPost.status === 'pending'`
2. `this.props.savedPost.status === 'draft' && this.props.post.status === 'pending'`

As noted in the original report, the issue pops up when a post is saved as pending (thus `this.props.savedPost.status==='pending'`) but the user wants to toggle pending off. The user has to toggle the option but an update to the toggle is only shown after save when the saved post is set back to another status.

This updates the comparison to set `isPending` to `true` if either:

1. `this.props.savedPost.status === 'pending' && this.props.post.status === 'pending'`
2. `this.props.savedPost.status === 'draft' && this.props.post.status === 'pending'`

Now, when a post is set to pending and a user toggles it off, `this.props.post.status` won't equal pending so the toggle responds correctly.

Note: Reference to `this.props.savedPost` in these comparisons was introduced back in 9196-gh-calypso-pre-oss to avoid showing UI changes optimistically.

Note 2: In the bug report, the question of "Should the auto-save kick in when this toggle is changed either direction?" was asked. Currently, `this.debouncedAutosave()` is only triggered when [the title changes](https://github.com/Automattic/wp-calypso/blob/master/client/post-editor/post-editor.jsx#L349) or [post content is updated](https://github.com/Automattic/wp-calypso/blob/master/client/post-editor/post-editor.jsx#L536). I'm not sure if we want to, but I didn't address that here.

## Testing
1. Fire up this branch
2. Open a new post
3. Type in some content. 
4. Toggle "Pending" on and save the post. The toggle should appear checked.
5. Toggle "Pending" off. The toggle should appear unchecked immediately instead of requiring a save.

## GIF
pending

ESLint was throwing some errors so I also updated so semicolons.

### CODE REVIEW

**Comment by timmyc** [comment]
Thanks for the PR @jeremeylduvall - what do you think about maybe adding an `isPending` method to [post/utils](https://github.com/Automattic/wp-calypso/blob/master/client/lib/posts/utils.js) and using that here?

**Comment by jeremeylduvall** [comment]
Thanks @timmyc! Added `isPending` to posts/utils and used that method to determine whether the post is pending. I also added three test scenarios in posts/test/utils.js. 

Currently, I'm just passing `this.props.post` into `isPending`. Is that sufficient? Previously, we were checking against `this.props.savedPost` as well.

File: client/post-editor/edit-post-status/index.jsx [hunk 1]
```diff
@@ -107,9 +107,7 @@ const EditPostStatus = React.createClass( {
 
 		if ( this.props.post ) {
 			isSticky = this.props.post.sticky;
-			isPending = this.props.savedPost.status === 'pending' || (
-				this.props.savedPost.status === 'draft' &&
-				this.props.post.status === 'pending' );
+			isPending = postUtils.isPending( this.props.post );
```
Comment (timmyc) [inline client/post-editor/edit-post-status/index.jsx]: So to truly do the same logic as you had before, this check should be against `savedPost` here, and the other clause could be changed to ` || ( this.props.savedPost.status === 'draft' && postUtils.isPending( this.props.post) `.  I suppose we could also add an `isDraft` utility as I'm guessing there are quite a few checks like this elsewhere.  But could be done in a different PR.

Looking just a few lines down, there are also some refactoring options to use postUtils more in this component.

**Comment by jeremeylduvall** [comment]
Chatted about this a bit in Slack. I'm not sure whether we need to reference `this.props.savedPost.status` when determining whether `isPending` should be set to `true` or if we can just reference `this.props.post`. 

The original issue was that we were including a condition for `this.props.savedPost.status === 'pending'`, which can be `true` even when the user toggles pending off for the current post.

If we still want to reference `this.props.savedPost` as the original implementation, we could go with:

```
isPending = ( postUtils.isPending( this.props.savedPost ) && 
	postUtils.isPending( this.props.post) ) || (
	this.props.savedPost.status === 'draft' &&
	postUtils.isPending( this.props.post ) );
```

If that's not necessary, we can go with the current branch, which only compares against `this.props.post` (similar to how we determine `isSticky`).

**Comment by timmyc** [comment]
Working well for me in testing so disregard my comment above!

Thanks again for the fix!

**Comment by jeremeylduvall** [comment]
Thanks!

**Pull request merged**